From ccce76d2850d6e52da3d7986c950af068fbfe0fd Mon Sep 17 00:00:00 2001
From: Arthur Gautier <arthur.gautier@arista.com>
Date: Sat, 13 Dec 2025 20:07:11 -0800
Subject: [PATCH] fix musl builds

This commit fixes the build on musl which fails with the following
error:
```
oslib/linux-blkzoned.c: In function 'blkzoned_move_zone_wp':
oslib/linux-blkzoned.c:389:37: error: 'FALLOC_FL_ZERO_RANGE' undeclared (first use in this function)
  389 |                 ret = fallocate(fd, FALLOC_FL_ZERO_RANGE, z->wp, length);
      |                                     ^~~~~~~~~~~~~~~~~~~~
oslib/linux-blkzoned.c:389:37: note: each undeclared identifier is reported only once for each function it appears in
make: *** [Makefile:501: oslib/linux-blkzoned.o] Error 1
make: *** Waiting for unfinished jobs....
```

Signed-off-by: Arthur Gautier <arthur.gautier@arista.com>

Upstream: https://github.com/axboe/fio/commit/ccce76d2850d6e52da3d7986c950af068fbfe0fd

Signed-off-by: Bernd Kuhls <bernd@kuhls.net>
---
 oslib/linux-blkzoned.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/oslib/linux-blkzoned.c b/oslib/linux-blkzoned.c
index 78e25fca91..c45ef623de 100644
--- a/oslib/linux-blkzoned.c
+++ b/oslib/linux-blkzoned.c
@@ -25,6 +25,7 @@
 #ifndef BLKFINISHZONE
 #define BLKFINISHZONE _IOW(0x12, 136, struct blk_zone_range)
 #endif
+#include <linux/falloc.h>
 
 /*
  * If the uapi headers installed on the system lacks zone capacity support,
